java/arcs/core/analysis/ClaimDeduction.kt

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+package arcs.core.analysis
+
+import arcs.core.data.AccessPath
+import arcs.core.data.expression.Expression
+
+internal typealias Path = List<AccessPath.Selector.Field>
+internal typealias Paths = List<Path>
+internal typealias ClaimDerivations = Map<Path, Set<Path>>
```

Comment:
This doesn't take into account Piotr's question about scopes introduces in expressions?